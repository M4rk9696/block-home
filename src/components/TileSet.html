<div class="page-tile-set">
  {#each sites as site, idx}
    <Tile tile={site} on:removeTile="handleRemove(idx)" on:select="handleClick(idx, site.url)"/>
  {:else}
    <Tile tile=""/>
  {/each}
</div>

<script>
  export default {
    components: {
      Tile: './Tile.html'
    },
    methods: {
      handleRemove(idx) {
        event.stopPropagation();
        this.store.deleteSite(idx);
        // this.store.saveSites();
      },
      handleClick(idx, url) {
        if(!url || url === '') {
          this.store.set({
            showQuickConfig: true,
            quickConfigIdx: idx,
            quickConfigSite: this.store.get().sites[idx]
          });
        }
        else {
          window.location = url;
        }
      }
    }
  };
</script>

<style>
  .page-tile-set {
    display: flex;
    flex-wrap: wrap;
    flex-direction: row;
    margin: 20px;
    padding: 0;
    font-size: 1.3rem;
  }
  @media only screen and (min-width: 1000px) {
    .page-tile-set {
      width: 60vw;
    }
  }
</style>
