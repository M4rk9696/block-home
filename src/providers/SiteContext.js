import React, { createContext, useState } from 'react'
import { saveSites } from './ChromeAPI'

const SiteContext = createContext({})

export const SiteProvider = ({ children }) => {
  const [sites, setSites] = useState([])
  const save = async () => saveSites(sites)

  return (
    <SiteContext.Provider
      value={{
        save,
        sites,
        setSites,
      }}
    >
      {children}
    </SiteContext.Provider>
  )
}

export default SiteContext
