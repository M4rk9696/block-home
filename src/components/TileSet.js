import React from 'react'
import styled from 'styled-components'
import Tile from './Tile'
import SiteContext from '../providers/SiteContext'

const TileSet = () => {
  return (
    <PageTileSet>
      <SiteContext.Consumer>
        {tiles => (
          <For each="tile" of={tiles} index="index">
            <Tile
              key={index}
              character={tile.character}
              url={tile.url}
              color={tile.color}
              title={tile.title}
            />
          </For>
        )}
      </SiteContext.Consumer>
    </PageTileSet>
  )
}

const PageTileSet = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;
  margin: 20px;
  padding: 0;
  font-size: 1.3rem;

  @media only screen and (min-width: 1000px) {
    .page-tile-set {
      width: 60vw;
    }
  }
`

export default TileSet
