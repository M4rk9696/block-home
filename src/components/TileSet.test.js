import React from 'react'
import TileSet from './TileSet'
import SiteContext from '../providers/SiteContext'
import {render} from '@testing-library/react'

describe('should render TileSet', () => {
  it('should be able to mount the component', () => {
    const sites = [
      {
        title: 'Dummy1',
        url: 'dummy_url_1',
        character: 'D_1',
        color: '#f3f3f3',
      },{
        title: 'Dummy2',
        url: 'dummy_url_2',
        character: 'D_2',
        color: '#f2f2f2',
      },
    ];
    const {container} = render(
      <SiteContext.Provider value={sites}>
        <TileSet />
      </SiteContext.Provider>
    )

    expect(container).toMatchSnapshot()
  })
})
