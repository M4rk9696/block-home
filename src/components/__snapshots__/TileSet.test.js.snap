// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TileSet should be able to mount the component 1`] = `
<div>
  <div
    class="sc-EHOje eqfsHd"
  >
    <div
      class="sc-ifAKCX dEMnFe"
    >
      <div
        class="sc-bxivhb WGaQA"
        color="#f3f3f3"
      >
        <div
          class="top-buttons"
        >
          <button
            class="sc-bdVaJa iibjPH"
          >
            ✖
          </button>
        </div>
        <span
          class="sc-bwzfXH iyaRUM"
        >
          D_1
        </span>
        <span
          class="sc-htpNat fPitBk"
        >
          Dummy1
        </span>
      </div>
    </div>
    <div
      class="sc-ifAKCX dEMnFe"
    >
      <div
        class="sc-bxivhb klfMum"
        color="#f2f2f2"
      >
        <div
          class="top-buttons"
        >
          <button
            class="sc-bdVaJa iibjPH"
          >
            ✖
          </button>
        </div>
        <span
          class="sc-bwzfXH iyaRUM"
        >
          D_2
        </span>
        <span
          class="sc-htpNat fPitBk"
        >
          Dummy2
        </span>
      </div>
    </div>
  </div>
</div>
`;
